kt/godot-runtime/src/main/kotlin/godot/core/GarbageCollector.kt

```diff
@@ -254,9 +254,16 @@ object GarbageCollector {
     }
 
     fun cleanUp() {
-        for (instance in staticInstances) {
-            instance.collect()
+        info("Start collecting static instances")
+        while(staticInstances.size > 0){
+            val iterator = staticInstances.iterator()
+            staticInstances = mutableSetOf()
+            for (instance in iterator) {
+                println("Collect ${instance::class}")
+                instance.collect()
```

Comment:
We use GD singleton to check if instance is valid. What would happen if GD singleton is collected before other singleton ?

Reply:
No longer an issue with the new delegate system.